R/tostring.R

```diff
@@ -1,3 +1,244 @@
+## state completely sucks and I hate it, but
+## we need a pdf device open to calculate the
+## print width of strings, and we can't be opening
+## a new one every time we want to
+font_dev_state <- new.env()
+font_dev_state$open <- FALSE
+font_dev_state$fontspec <- list()
+font_dev_state$spacewidth <- NA_real_
+font_dev_state$ismonospace <- NA
+font_dev_state$max_ratio <- NA_real_
+font_dev_state$dev_num <- NA_integer_
+
```

Comment:
I understand the underlying need for this. I am only wondering if there is an alternative. Maybe @pawelru has an alternative?

Reply:
If this was covered sufficiently in ourmeeting, please resolve it, otherwise we can discuss more on wed